ops/cli.py

```diff
@@ -1,351 +1,309 @@
+    success(f"{config.command} is done!")
+    info(shell("kubectl get pods"), plain=True)
+
+
+command_mapping: Dict[str, List[Callable[[WflInstanceConfig], None]]] = {
+    "info": [
+        infer_missing_arguments_pre_validate,
+        validate_cloud_sql_name,
+        validate_cluster_name,
+        infer_missing_arguments,
+        print_config
+    ],
+    "connect": [
+        infer_missing_arguments_pre_validate,
+        validate_cloud_sql_name,
+        infer_missing_arguments,
+        print_config,
+        exit_if_dry_run,
+        configure_cloud_sql_proxy,
+        print_cloud_sql_proxy_instructions
+    ],
+    "deploy": [
+        infer_missing_arguments_pre_validate,
+        validate_cloud_sql_name,
+        validate_cluster_name,
+        infer_missing_arguments,
+        print_config,
+        configure_kubectl,
+        configure_helm,
+        render_values_file,
+        exit_if_dry_run,
+        prompt_deploy_version,
+        configure_cloud_sql_proxy,
+        publish_docker_images,
+        helm_deploy_wfl,
+        run_liquibase_migration,
+        stop_cloud_sql_proxy,
+        print_deployment_success
+    ]
+}
+
+
+def cli() -> WflInstanceConfig:
+    """Configure the arguments, help text, and parsing."""
+    parser = argparse.ArgumentParser(description="deploy or connect to WFL infrastructure",
+                                     usage="%(prog)s [-h] [-d] COMMAND ENV INSTANCE [...]")
+    parser.add_argument("command", choices=command_mapping.keys(), metavar="COMMAND",
+                        help=f"one of [{', '.join(command_mapping.keys())}]")
+    parser.add_argument("environment", choices=["gotc-dev", "gotc-prod", "aou"], metavar="ENV",
+                        help="specify 'gotc-deploy/deploy/{ENV}' with one of [%(choices)s]")
+    parser.add_argument("instance_id", metavar="INSTANCE",
+                        help="specify the ID of the instance to use in the environment")
+    parser.add_argument("-d", "--dry-run", action="store_true",
+                        help="exit before COMMAND makes any remote changes")
+    parser.add_argument("-v", "--version",
+                        help="specify the version to use instead of the 'version' file")
+    parser.add_argument("--project",
+                        help="specify the GCP project name instead of 'broad-{ENV}'")
+    parser.add_argument("--cloud-sql-name",
+                        help="specify the Cloud SQL name instead of inferring from labels")
+    parser.add_argument("--cluster-zone",
+                        help="specify the zone for the GKE cluster instead of inferring from CLUSTER_NAME")
+    parser.add_argument("--cluster-name",
+                        help="specify the GKE cluster name instead of inferring from labels")
+    parser.add_argument("--cluster-namespace",
+                        help="specify the K8s namespace instead of '{INSTANCE}-wfl'")
+    return WflInstanceConfig(**vars(parser.parse_args()))
```

Comment:
Match argparse keys to `@dataclass` constructor keys (which are the same as the WflInstanceConfig fields)